pd/src/info/specific.rs

```diff
@@ -429,6 +495,80 @@ impl SpecificQueryService for Info {
         }
     }
 
+    #[instrument(skip(self, request))]
+    async fn swap_executions(
+        &self,
+        request: tonic::Request<SwapExecutionsRequest>,
+    ) -> Result<tonic::Response<Self::SwapExecutionsStream>, Status> {
+        let state = self.storage.latest_snapshot();
+        state
+            .check_chain_id(&request.get_ref().chain_id)
+            .await
+            .map_err(|e| tonic::Status::unknown(format!("chain_id not OK: {e}")))?;
+        let request_inner = request.into_inner();
+        let start_height = request_inner.start_height;
+        let end_height = request_inner.end_height;
+        let trading_pair = request_inner.trading_pair;
+
+        // Convert to domain type ahead of time if necessary
+        let trading_pair: Option<TradingPair> = if let Some(trading_pair) = trading_pair {
+            Some(trading_pair.try_into().expect("invalid trading pair"))
```

Comment:
I think we don't want to `expect` here, since then someone can crash the node by submitting a request with an invalid trading pair.